pkg/extensions/extension.go

```diff
@@ -16,7 +18,7 @@ type Extension interface {
 //go:generate moq -rm -out manager_mock.go . ExtensionManager
 type ExtensionManager interface {
 	List(includeMetadata bool) []Extension
-	Install(url string, stdout, stderr io.Writer) error
```

Comment:
The fact that `Install()` now doesn't accept stdout/stderr and instead stores it internally via iostreams created a slight asymmetry with the signature of `Upgrade()`, which still takes stdout & stderr via arguments. I feel that it was not necessary for the addition of the binary extensions feature to change the logic around iostreams, and that stdout/stderr could have remained as arguments.

I do agree that passing in repo as `ghrepo.Interface` instead of a clone URL is better because it frees the Manager from having to parse out URLs internally to determine the repository 👍

Reply:
I was going to make the interfaces consistent in subsequent PRs (and have already updated `Upgrade` in my branch that's stack on this one).